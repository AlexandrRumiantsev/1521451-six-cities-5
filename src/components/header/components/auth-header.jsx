import React from 'react';
import {
  Link,
  Router
} from "react-router-dom";
import {appUser} from "../../../shapes/app-user";


import browserHistory from "../../../browser-history";

const AuthHeader = ({user}) => {
  return (
    <div className="header__nav-link header__nav-link--profile">
      <img className="header__avatar-wrapper" src={user.data.avatarUrl} />
      <span className="header__user-name user__name">
        <Router history={browserHistory}>
          <Link to='/favorites'> {user.data.email} </Link>
        </Router>
      </span>
    </div>
  );
};

AuthHeader.propTypes = {
  user: appUser
};

export default AuthHeader;
