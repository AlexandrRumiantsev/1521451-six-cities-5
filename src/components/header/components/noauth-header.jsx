import React from 'react';
import {
  Link,
  Router
} from "react-router-dom";

const newHistory = require("history").createBrowserHistory();

const NotAuthHeader = () => {
  return (
    <div className="header__nav-link header__nav-link--profile">
      <div className="header__avatar-wrapper user__avatar-wrapper">
      </div>
      <span className="header__user-name user__name">
      <Router history={newHistory}>
         <Link to='/login'> Sign In </Link>
      </Router>
      </span>
    </div>
  );
};

export default NotAuthHeader;
