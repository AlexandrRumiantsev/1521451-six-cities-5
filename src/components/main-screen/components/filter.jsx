import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {offerItem} from '../../../shapes/offer-item';


const enhanceWithClickOutside = require('react-click-outside');

class Filter extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      select: ``
    }

    this.filterDOM = React.createRef();
    this.filterToggle = this.filterToggle.bind(this);
    this.filterChange = this.filterChange.bind(this);
  }
  filterChange(event) {
    const {currentOffers} = this.props;
    filterOffer(
        event.target.innerText,
        currentOffers
    );
  }
  handleClickOutside() {
    this.filterToggle();
  }
  filterToggle() {
    (this.state.select == `--opened`) ? this.setState({
      select: ``
    }) : this.setState({
      select: `--opened`
    })
  }
  render() {
    return (
      <React.Fragment>
        <form className="places__sorting" action="#" method="get">
          <span className="places__sorting-caption">Sort by</span>
          <span className="places__sorting-type" onClick={this.filterToggle} tabIndex="0">
            {this.props.baseFilter}
            <svg className="places__sorting-arrow" width="7" height="4">
              <use xlinkHref="#icon-arrow-select"></use>
            </svg>
          </span>
          <ul ref={this.filterDOM} className={`places__options places__options--custom places__options`+this.state.select}>
            <li onClick={this.filterChange} className="places__option places__option--active" tabIndex="0">Popular</li>
            <li onClick={this.filterChange} className="places__option" tabIndex="0">Price: low to high</li>
            <li onClick={this.filterChange} className="places__option" tabIndex="0">Price: high to low</li>
            <li onClick={this.filterChange} className="places__option" tabIndex="0">Top rated first</li>
          </ul>
          <select defaultValue={`Popular`} className="places__sorting-type" id="places-sorting">
            <option className="places__option" value="popular">Popular</option>
            <option className="places__option" value="to-high">Price: low to high</option>
            <option className="places__option" value="to-low">Price: high to low</option>
            <option className="places__option" value="top-rated">Top rated first</option>
          </select>
        </form>
      </React.Fragment>
    );
  }
}

Filter.propTypes = {
  filterOffer: PropTypes.func,
  baseFilter: PropTypes.string,
  currentOffers: PropTypes.arrayOf(
      PropTypes.shape({
        offerItem
      })
  ),
};

export default enhanceWithClickOutside(Filter);
