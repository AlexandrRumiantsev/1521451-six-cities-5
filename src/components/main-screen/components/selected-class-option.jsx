import React from 'react';
import PropTypes from 'prop-types';
import {offerItem} from '../../../shapes/offer-item';


const SelectedClassOption = (props) => {
  const {optionList, baseFilter} = props;

  const option = (props, type) => {
    switch(type){
      case `li`:
        return (<li data-action="change"
          key={`option-${props}`}
          className={`places__option ${(baseFilter == props) ? `places__option--active` : ``}`}
          tabIndex="0">
          {props}
        </li>)
       case `option`:
        return (
          <option className="places__option" value="popular">{props}</option>
        )
    }

    }
    const createSelectArray = (object, type) => {
      const array = [];
      for (const property in object) {
          array.push(
            option(object[property], type)
          );
      }
      return array;
    }

  return (
    <form onClick={props.click} className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span className="places__sorting-type" data-action="open" tabIndex="0">
        {props.baseFilter}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"></use>
        </svg>
      </span>
      <ul className={
        `places__options places__options--custom places__options${props.select ? props.select : ``}`
      }
      >
        {
          createSelectArray(optionList , 'li')
        }
      </ul>
      <select defaultValue={`Popular`} className="places__sorting-type" id="places-sorting">
        {
          createSelectArray(optionList , 'option')
        }
      </select>
    </form>
  );
};


SelectedClassOption.propTypes = {
  filterOffer: PropTypes.func,
  baseFilter: PropTypes.string,
  currentOffers: PropTypes.arrayOf(
      PropTypes.shape({
        offerItem
      })
  ),
  optionList: PropTypes.arrayOf(
      PropTypes.string
  ),
  click: PropTypes.func,
  select: PropTypes.string,
};

export default SelectedClassOption;
