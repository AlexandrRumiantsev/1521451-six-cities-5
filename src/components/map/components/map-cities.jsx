import React, {PureComponent} from 'react';
import {Map, TileLayer} from "react-leaflet";
//import {offerItem} from '../../shapes/offer-item';
import PropTypes from 'prop-types';
import L from 'leaflet';


const city = [52.38333, 4.9];

class MapCities extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      map: ''
    }

  }
  filterOffers(offers, currentCity, cities){
    return {
      offers: offers.filter(
        offer => offer.city == currentCity
      ),
      currentCityInfo: cities.filter(
        city => city.title == currentCity
      )[0]
    }
  }
  settingsMap(info){
    const icon = L.icon({
      iconUrl: `/img/pin.svg`,
      iconSize: [30, 30]
    });

    const zoom = 12;
    this.map = L.map(`map`, {
      center: [info['currentCityInfo'].lat, info['currentCityInfo'].lon],
      zoom,
      zoomControl: false,
      marker: true
    });
    this.map.setView([info['currentCityInfo'].lat, info['currentCityInfo'].lon], zoom);

    L.tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
      attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
    })
      .addTo(this.map);


      info['offers'].map((city) => {
        const offerCords = [city.map.lat, city.map.lon];
        L.marker(offerCords, {icon}).addTo(this.map);
      })
  }
  componentDidUpdate(prevProps) {
    console.log(`componentDidUpdate ${this.props.currentCity}`);
/*Вынести в отдельную функцию с параметрами
    if(this.props.currentCity !== prevProps.currentCity){

      console.log(`prevProps ${prevProps.currentCity}`);



      let cites = this.props.offers.filter(
        offer => offer.id !=  window.location.href.split('/')[4] && offer.city == this.props.currentCity
      );


      this.map.remove();
      console.log(cites);
      const icon = L.icon({
        iconUrl: `/img/pin.svg`,
        iconSize: [30, 30]
      });

      const zoom = 12;
      const map = L.map(`map`, {
        center: [info[0].lat, info[0].lon],
        zoom,
        zoomControl: false,
        marker: true
      });
      this.map = map;
      map.setView([info[0].lat, info[0].lon], zoom);

      L.tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
        .addTo(map);


        cites.map((city) => {
          const offerCords = [city.map.lat, city.map.lon];
          L.marker(offerCords, {icon}).addTo(map);
        })
        /
    }*/
  }
  componentDidMount() {
    const {currentCity, offers, cities} = this.props;
    this.settingsMap(
      this.filterOffers(
        offers,
        currentCity,
        cities
      )
    )
    /*let cites = offers.filter(
      offer => offer.id !=  window.location.href.split('/')[4] && offer.city == currentCity
    );*/



  }
  render() {
    return (

      <div id='map'>
        <Map
          doubleClickZoom={true}
          scrollWheelZoom={true}
          dragging={true}
          animate={true}
          easeLinearity={0.35}
        >
          <TileLayer
            attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
            url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
          />
        </Map>
        </div>

    );
  }
}
/*
MapCity.propTypes = {
  offers: PropTypes.arrayOf(
      PropTypes.shape({
        offerItem
      })
  )
};*/

export {MapCities};
