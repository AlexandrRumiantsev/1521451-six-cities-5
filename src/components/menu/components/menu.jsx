import React from 'react';
import {
  Link
} from "react-router-dom";
import {citiesShape} from '../../../shapes/citiesShape';
import PropTypes from 'prop-types';

const Menu = ({currentCity, changeCity, activeItem, sortOffers}) => {

  return (<ul className="locations__list tabs__list">
      {Array.from(sortOffers).map((value) => {
        return(
          <li key={value[0]} className="locations__item">
            <Link onClick={
              () => changeCity(value[0])
            }
            className={`locations__item-link tabs__item${(value[0] === currentCity) ? activeItem : ``}`}
            to={value[0]}
            >
              <span>{value[0]}</span>
            </Link>
          </li>
        )

      })}
    </ul>)
};

Menu.propTypes = {
  currentCity: PropTypes.string,
  changeCity: PropTypes.func,
  activeItem: PropTypes.string,
  cities: PropTypes.arrayOf(
      PropTypes.shape({
        citiesShape
      })
  ),
};

export default Menu;
