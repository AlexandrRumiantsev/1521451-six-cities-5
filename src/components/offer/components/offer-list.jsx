import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import OfferItem from './offer-item.jsx';


class OfferList extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {offers, currentCity, param, max, currentOffer} = this.props;

    const currentParam = (param) ? param : currentCity;

    const offersFiltred = offers.filter(
      offer => offer.city == currentParam && offer.id != currentOffer
    )


    return (
      <div className="near-places__list places__list">
      {
        offersFiltred.map((offer, index) => {
          if(max){
            if(index < max){
              return (<OfferItem key={offer.id} offer={offer} />);
            }
          }else{
            return (<OfferItem key={offer.id} offer={offer} />);
          }
        })
      }
    </div>
    );
  }
}

/*
OfferList.propTypes = {
  offers: PropTypes.arrayOf(
      PropTypes.shape({
        offerItem
      })
  ),
  max: PropTypes.number,
  currentOffer: PropTypes.number
};
*/


export default OfferList
