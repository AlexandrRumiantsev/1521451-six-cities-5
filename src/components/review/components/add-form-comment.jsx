import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

class AddFormComment extends PureComponent {
  constructor(props) {
    super(props);
  }
  render() {
    const {
      titleInputs,
      handleSubmit,
      handleChange,
      inputRating,
      ratingField,
      textField,
      refText,
      refRating
    } = this.props;

    return (
      <form disabled onSubmit={handleSubmit} className="reviews__form form" action="#" method="post">
        <label className="reviews__label form__label" htmlFor="review">Your review</label>
        <div className="reviews__rating-form form__rating">
          {titleInputs.map((item, index) => (
            <React.Fragment key={item}>
              {inputRating(index, item, titleInputs, refRating, handleChange)}
            </React.Fragment>
          ))}
        </div>
        <textarea ref={refText} data-type="comment" onInput={(e) => handleChange(e)} className="reviews__textarea form__textarea" id="review" name="review" placeholder="Tell how was your stay, what you like and what can be improved"></textarea>
        <div className="reviews__button-wrapper">
          <p className="reviews__help">
            To submit review please make sure to set
            <span className="reviews__star">
              rating
            </span>
            and describe your stay with at least
            <b className="reviews__text-amount">
              50 characters
            </b>.
          </p>
          <button
            className="reviews__submit form__submit button"
            disabled={!(ratingField && textField)}
            type="submit">
            Submit
          </button>
        </div>
      </form>
    );
  }
}

AddFormComment.propTypes = {
  addReviews: PropTypes.func,
  currentOffer: PropTypes.number,
  titleInputs: PropTypes.arrayOf(
      PropTypes.string
  ),
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func,
  inputRating: PropTypes.func,
  ratingField: PropTypes.bool,
  textField: PropTypes.bool,
  refText: PropTypes.object,
  refRating: PropTypes.object,
};

export default AddFormComment;
