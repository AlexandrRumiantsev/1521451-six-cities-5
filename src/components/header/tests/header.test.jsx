import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import HeaderContainer from "../header-container";
import {authHeader} from "../components/auth-header";
import {notAuthHeader} from "../components/noauth-header";;

import {userData} from "./mocks";


describe(`Render Header`, () => {

  it(`Render authHeader`, () => {
    const mockStore = configureStore([]);
    let authHeaderRenderer = null;
    let store = null;

    beforeEach(() => {
      store = mockStore({
        user: {
          data: userData,
          authorizationStatus: `AUTH`
        }
      });

      store.dispatch = jest.fn();

      authHeaderRenderer = renderer
      .create(
        <Provider store={store}>
          <authHeader />
        </Provider>
      )
      .toJSON();
      });

  })


  it(`Render notAuthHeader`, () => {
    const mockStore = configureStore([]);
    let notAuthHeaderRenderer = null;
    let store = null;

    beforeEach(() => {
      store = mockStore({
        user: {
          data: userData,
          authorizationStatus: `AUTH`
        }
      });

      store.dispatch = jest.fn();

      notAuthHeaderRenderer = renderer
      .create(
        <Provider store={store}>
          <notAuthHeader />
        </Provider>
      )
      .toJSON();
      });


  })


})
