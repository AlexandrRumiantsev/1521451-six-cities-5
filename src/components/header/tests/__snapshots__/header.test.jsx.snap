// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Render HeaderContainer 1`] = `null`;

exports[`Render Header Render authHeader 1`] = `<authHeader />`;

exports[`Render Header Render notAuthHeader 1`] = `<notAuthHeader />`;
